### PR TITLE
[DROOLS-6676] Updated archetypes. Fixed "archetype:integration-test" goal

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @jiripetrlik 

**Thank you for submitting this pull request**

**NOTE!:** Double check the target branch for this PR.
The default is `main` so it will target Drools 8 / Kogito / Optaplanner 8.
If this PR is not strictly related to drools and kogito project in `drools.git`, it should probably target `7.x`as a branch

**Ports** If a forward-port or a backport is needed, paste the forward port PR here

[link](https://www.example.com)

**JIRA**: 

[link](https://issues.redhat.com/browse/DROOLS-6676)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-evaluator/pom.xml [hunk 1]
```diff
@@ -28,15 +28,10 @@
...
-      <dependency>
-        <groupId>org.kie</groupId>
-        <artifactId>kie-pmml-evaluator-core</artifactId>
-        <version>${version.org.kie}</version>
-      </dependency>
       <dependency>
         <groupId>org.kie</groupId>
         <artifactId>kie-pmml-models-naivebayes-model</artifactId>
-        <version>${version.org.kie}</version>
+        <version>\${version.org.kie}</version>
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-evaluator/pom.xml:34]: Why this change?

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-model/src/main/java/org/kie/pmml/models/naive_bayes/model/KiePMMLNaiveBayesModel.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ * Copyright ${year} Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-model/src/main/java/org/kie/pmml/models/naive_bayes/model/KiePMMLNaiveBayesModel.java:2]: This should be the year I guess

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources/projects/generated-naivebayes/goal.txt [hunk 3]
```diff
@@ -1 +1 @@
-verify
\ No newline at end of file
+compile
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources/projects/generated-naivebayes/goal.txt:1]: Why this change?

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-model/src/main/java/org/kie/pmml/models/naive_bayes/model/KiePMMLNaiveBayesModel.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-model/src/main/java/org/kie/pmml/models/naive_bayes/model/KiePMMLNaiveBayesModel.java:2]: @danielezonca 
? It is a "filtered" resource

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources/projects/generated-naivebayes/goal.txt (hunk 3 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources/projects/generated-naivebayes/goal.txt:1]: @danielezonca 
Because actual integration test does not works, since generated methods trows "UnsupportedOperationException"

File: kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-evaluator/pom.xml (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models-archetype/src/test/resources-filtered/projects/generated-naivebayes/reference/kie-pmml-models-naivebayes-evaluator/pom.xml:34]: @danielezonca 
because it must be "unescaped" - i.e. in the generated file it must be " <version>${version.org.kie}</version>" (without the slash, it would resolve to the current kie version, that is wrong)

**Review by danielezonca — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources-filtered/projects/generated-tree/reference/kie-pmml-models-drools-tree-tests/pom.xml [hunk 4]
```diff
@@ -64,23 +89,17 @@
       <dependency>
         <groupId>org.kie</groupId>
         <artifactId>kie-pmml-models-drools-tree-model</artifactId>
-        <version>${version.org.kie}</version>
+        <version>\${version.org.kie}</version>
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources-filtered/projects/generated-tree/reference/kie-pmml-models-drools-tree-tests/pom.xml:92]: Is this intentional?

File: kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources/projects/generated-tree/goal.txt [hunk 5]
```diff
@@ -1 +1 @@
-verify
\ No newline at end of file
+compile
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources/projects/generated-tree/goal.txt:1]: What is the reason for this change?

**Review by jiripetrlik — APPROVED** [review]
Looks good to me. I have two comments.

File: kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources-filtered/projects/generated-tree/reference/kie-pmml-models-drools-tree-tests/pom.xml (hunk 4 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources-filtered/projects/generated-tree/reference/kie-pmml-models-drools-tree-tests/pom.xml:92]: @jiripetrlik 
Yes. This file is "filtered". Withouth the `\`, `${version.org.kie}` would be resolved to current version. But in that specific place, the generated file must have exactly `<version>${version.org.kie}</version>`, and the trailing `\` allows that.

File: kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources/projects/generated-tree/goal.txt (hunk 5 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models-drools-archetype/src/test/resources/projects/generated-tree/goal.txt:1]: @jiripetrlik 
Because tests in the generated code are actually broken, since all the generated stub methods throws `UnsupportedOperationException`.
The scope of the archetype testing is to verify that actually compiles. This, together with equality to "reference" projects, should reasonably verify that archetype is generating a project suited for the current version: does this make sense ?
